### ISSUE TITLE
[WelcomeUX] Welcome page should not show during extension development/debugging

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#19569]

**Comment by chrmarti** [comment]
The setting should be enough to control this. Also: It doesn't open if there is any other editor opened/restored.

**Comment by Tyriar** [comment]
I just tested again on my home computer and it appears what I was running into was non-folder workspaces getting files restored but the welcome page still appears. Here's what appears when I debug:

![image](https://cloud.githubusercontent.com/assets/2193314/22607572/4d643886-ea0d-11e6-8129-8a4b1f6dcf65.png)

Closing welcome, opening another file and relaunching results in the same thing:

![image](https://cloud.githubusercontent.com/assets/2193314/22607603/69518f94-ea0d-11e6-8fd6-8dfd3d602148.png)

In other words, the welcome page should not show up when file(s) are restored.

**Comment by chrmarti** [comment]
@Tyriar Note the untitled file. Both that and Welcome use backupFileService.hasBackups() to check if there are backups that will be restored. Somehow that must return false in this case, could you look into this?

**Comment by Tyriar** [comment]
I believe these files are being restored by `EditorStacksModel.load` (using local storage key `storage://workspace/__$noWorkspace__editorstacks.model`). While files are not meant to be restored in non-folder workspaces until https://github.com/Microsoft/vscode/issues/207 is handled, maybe extension development was an unconsidered edge case? @bpasero should know more about this.

Regardless of the cause of the restored state, I think we should only show the welcome screen when no other files have been restored. So fixing this and https://github.com/Microsoft/vscode/issues/19689 by ensuring 0 tab have been created as well as no backup for the workspace exist (since they may not be visible initially) is probably the right fix here imo.

**Comment by bpasero** [comment]
Yes, extension development restores files even for the empty workspace case because I thought it would be very annoying otherwise to work on an extension always having to open the same file over and over again.
